desc/compute/utils.py

```diff
@@ -1066,8 +1104,13 @@ def surface_averages_map(grid, surface_label="rho", expand_out=True, tol=1e-14):
         ``function(q, sqrt_g)``.
 
     """
-    if not hasattr(grid, f"num_{surface_label}"):
-        expand_out = False  # don't try to expand already expanded output
+    surface_label = grid.get_label(surface_label)
+    expand_out = (
```

Comment:
In an older PR, jax threw some missing indices error that seemed to suggest surface integration was done on grid without unique idx defined. So we generalized these methods to work without them, with the cavet that we always expand out

Reply:
That's fine but I think then we should request the correct value for `expand_out` on the calling side and verify that we can expand and throw an error if the caller asked for expand_out but indices aren't given. As is this seems like it might silently change whether or not it expands out based on the type of grid.